src/RoomsSDKAdapter.js

```diff
@@ -159,6 +161,25 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
     return this.getRoomObservables[ID];
   }
 
+  /**
+   * Returns an observable that emits room data of the recently created room
+   * Observable will complete after one emission.
+   *
+   * @param {Room} room  Information about the room to create.
+   * @returns {external:Observable.<Room>} Observable stream that emits room data.
+   * @memberof RoomsAdapter
+   */
+  createRoom(room) {
+    logger.debug('ROOM', room, 'createRoom()', ['called with', {room}]);
+
+    return defer(() => this.datasource.rooms.create(room)).pipe(
```

Comment:
because this function returns an observable that emits a Room object, you have to make sure you return a Room object, not an SDK room object.

The mapping is currently done in the function "fetchRoom", but I would suggest factoring this simple mapping in a dedicated function called "fromSDKRoom". See how it's done for "fromSDKActivity", for example.

Reply:
Added a new private function "fromSDKRoom"